src/plugins/vis_type_timeseries/public/application/components/lib/get_click_filter_data.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { XYChartSeriesIdentifier, GeometryValue } from '@elastic/charts';
+import { ValueClickContext } from 'src/plugins/embeddable/public';
+import { X_ACCESSOR_INDEX } from '../../visualizations/constants';
+import { BUCKET_TYPES } from '../../../../common/enums';
+import { TimeseriesVisParams } from '../../../types';
+import type { TSVBTables } from './types';
+
+export const getClickFilterData = (
+  points: Array<[GeometryValue, XYChartSeriesIdentifier]>,
+  tables: TSVBTables,
+  model: TimeseriesVisParams
+) => {
+  const data: ValueClickContext['data']['data'] = [];
+  points.forEach((point) => {
+    const [geometry] = point;
+    const { specId } = point[1];
+    // specId for a split series has the format
+    // 61ca57f1-469d-11e7-af02-69e470af7417:Men's Accessories, <layer_id>:<split_label>
+    const termArray = specId.split(':');
+    const table = tables[termArray[0]];
+
+    const layer = model.series.filter(({ id }) => id === termArray[0]);
+    let splitLabel = termArray.length > 1 ? termArray[1] : undefined;
```

Comment:
Nit: Can you extract `termArray.length > 1` into a constant like `hasSplit` or something similar? I tripped over this a few times.

Reply:
I changed it to `const [layerId, splitLabel] = specId.split(':');`
and used the splitLabel instead. Are you ok with it?

https://github.com/elastic/kibana/pull/97426/commits/eb75702f27cd7652f6795cfadc5de553bcb41499